src/vs/workbench/contrib/debug/common/debugCompoundRoot.ts

```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IDebugCompoundRoot } from 'vs/workbench/contrib/debug/common/debug';
+import { Emitter } from 'vs/base/common/event';
+
+export class DebugCompoundRoot implements IDebugCompoundRoot {
+	private stopped = false;
+	private stopEmitter = new Emitter<void>();
+
```

Comment:
We usualy name our events like what happend, not what should happen as a consequence.
Thus I suggest the following name `onDidEndContainingSession` or `onDidEndChildSession` or just `onDidEndSession`. To be algined with the existing `onDidEndSession` that lives in the `debugService`